### ISSUE TITLE
Image Border Ghost Object Artefact

### ISSUE THREAD

**Issue by MicPhotonIGBMC** [mpicbg-csbd/stardist#116]
Good afternoon,

First of all, thank you for the great tool, we are big fan/users of StarDIst2D in our facility. 
I have been trying StarDist3D for some times now and very often I get this weird border artefact where StarDist get's confused by the noise and find object that do not exist. 

I have trained a network based on StarDist3D dataset https://github.com/mpicbg-csbd/stardist/releases/download/0.3.0/demo3D.zip (defaults parameters, 119+123 epochs training sessions via ZeroCostDL4Mic notebook)
It does also  happen with our own annotation. similar issue with real microscopy data and our own annotation. hat's intriguing is that the artefacts are always on the left or right border as you can with the prediction on the stack_028 and the same stack rotating 180 degrees to the right.

Any one with similar issue and advices?

thank you

Bertrand

![28_180r](https://user-images.githubusercontent.com/74547111/108389210-bd3e8880-720f-11eb-836c-9b7d5e0694f4.PNG)

![stack_028_StarDist3D](https://user-images.githubusercontent.com/74547111/108389191-b748a780-720f-11eb-8da9-40ca61bbd636.PNG)

**Comment by uschmidt83** [comment]
> First of all, thank you for the great tool, we are big fan/users of StarDIst2D in our facility.

Thanks, that's nice to hear!

> I have been trying StarDist3D for some times now and very often I get this weird border artefact where StarDist get's confused by the noise and find object that do not exist.
> 
> I have trained a network based on StarDist3D dataset https://github.com/mpicbg-csbd/stardist/releases/download/0.3.0/demo3D.zip (defaults parameters, 119+123 epochs training sessions via ZeroCostDL4Mic notebook)

I'm sorry, but I can't really comment on what the ZeroCostDL4Mic notebook does. It could be that the default parameters of the ZeroCostDL4Mic notebook are not suitable for the dataset that you're trying to use.

I can  tell you that this problem does not happen when training with the notebooks that [we provide](https://nbviewer.jupyter.org/github/mpicbg-csbd/stardist/tree/master/examples/3D/).

Best,
Uwe

**Comment by Romain-Laine** [comment]
Hi Bertrand and Uwe,
Thanks for pointing this out to us. I have now run another training session and indeed noticed the issue that you mentioned. That is not an issue that we had when we originally tested and released the notebook.

@MicPhotonIGBMC , would you like to file an issue on our GitHub page instead if that's a ZeroCost issue and not StarDist?
https://github.com/HenriquesLab/ZeroCostDL4Mic/issues

@uschmidt83 , were there any major changes to StarDist 3D in the past month that may have caused a change in behaviour that we should look into?

I will be testing these further in the coming days and will let you know.
Best,

Romain

**Comment by MicPhotonIGBMC** [comment]
Thank you
transferred as
Image Border Ghost Object Artefact #74
https://github.com/HenriquesLab/ZeroCostDL4Mic/issues/74

**Comment by uschmidt83** [comment]
Hi Romain,

> @uschmidt83 , were there any major changes to StarDist 3D in the past month that may have caused a change in behaviour that we should look into?

there hasn't been a new (official) release since November 2020 (see [releases](https://github.com/mpicbg-csbd/stardist/releases)).

Best,
Uwe

**Comment by Romain-Laine** [comment]
Hi Uwe,
Thanks for the info. It could be that. We haven't fully tested it for a while.
We have just had a go and it looks like running our notebooks on TF2.4 really helps stability (as opposed to 1.5) and is looking like it's fixing the problem. Is this something that you noticed too?
I will help Bertrand fix the problem with our notebook from our issue page from on.

Thanks!

Romain

**Comment by uschmidt83** [comment]
> We have just had a go and it looks like running our notebooks on TF2.4 really helps stability (as opposed to 1.5)

I haven't used TF 1.x for a while, but I didn't notice any "stability" improvements since StarDist was always quite stable to train for me anyways. What do you mean?

> and is looking like it's fixing the problem. Is this something that you noticed too?

Since I don't know this problem and we created this notebook while using TF 1.x, I wouldn't assume that think TF 2.x fixed it.

Uwe

**Comment by Romain-Laine** [comment]
Hi Uwe,
It's entirely possible it's specific to our implementation then. What I meant by unstable is that I had obtained a few times an unusual loss curve with an additional peak half way through the number of epoch. See below. This does not match any changes in learning rate either.

<img width="910" alt="Screenshot 2021-02-24 at 22 12 44" src="https://user-images.githubusercontent.com/21193399/109297054-4d408b80-7829-11eb-9aac-f0e9788e5907.png">

But either way, we've now upgraded it to TF2.x and implemented the default settings as close to yours as possible, as well as the exact ways to do augmentation. And this seems to perform much better and not give the phantom masks or the unusual loss curves.

Thanks a lot for your help. I really appreciate it.
Best,

Romain
